File: leetcode/1304-find-n-unique-integers-sum-up-to-zero/find-n-unique-integers-sum-up-to-zero.js
/**
 * @param {number} n
 * @return {number[]}
 */
var sumZero = function(n) {
  const result = [];
  const mid = n >>> 1;
  if (n % 2 === 1) {
    result[mid] = 0;
  }

  for (let i = 0, k = 1; i < mid; i++, k++) {
    result[i] = k;
    result[n - i - 1] = -k;
  }

  return result;
};

File: leetcode/1304-find-n-unique-integers-sum-up-to-zero/ FindNUniqueIntegersSumUpToZero.java
class Solution {
  public int[] sumZero(int n) {
    int[] result = new int[n];
    if (n % 2 == 0) {
      for (int i = 0; i < n / 2; i++) {
        result[i * 2] = i + 1;
        result[i * 2 + 1] = -(i + 1);
      }
    } else {
      result[0] = 0;
      for (int i = 0; i < n / 2; i++) {
        result[i * 2 + 1] = i + 1;
        result[i * 2 + 2] = -(i + 1);
      }
    }
    return result;
  }
}
